src/transport/unicast/link/task/read.c

```diff
@@ -89,12 +89,17 @@ void *_zp_unicast_read_task(void *ztu_arg) {
 
         if (ret == _Z_RES_OK) {
             ret = _z_unicast_handle_transport_message(ztu, &t_msg);
+            bool should_close = (_Z_MID(t_msg._header) == _Z_MID_CLOSE);
```

Comment:
The message inspection should be done by `_z_unicast_handle_transport_message` which in turn should return an appropriate error code. If `ret != _Z_RES_OK` then the task is already marked as non running at line `95` below.